src/platform_impl/macos/window.rs

```diff
@@ -79,6 +89,7 @@ pub struct PlatformSpecificWindowBuilderAttributes {
     pub disallow_hidpi: bool,
     pub has_shadow: bool,
     pub accepts_first_mouse: bool,
+    pub parent: Parent,
```

Comment:
I think I'd rather use `Option<Id<NSWindow, Shared>>` here, and then create the `Id<NSWindow, Shared>` directly inside `WindowBuilderExtMacOS::with_parent_window` by retaining the pointer with `Id::retain` (that way we also know that the window will be valid for the duration we need it to be)

Reply:
I can do this but `NSWindow` defined in `appkit` module isn't public.
Can I just re-export them to platform module?

Reply:
I think you can just do `pub(crate) parent: Option<Id<NSWindow, Shared>>`?